TestHelpers/struct_field_to_index_map_generator.py

```diff
@@ -0,0 +1,312 @@
+'''
+[A.x] => [A.0]
+[A.y] => [A.1]
+[B.x] => [B.0]
+[B.y] => [B.1]
+'''
+# this is used for Testers/Passes when we want to specify
+# a field of a variable by field name (rather than index)
+# to test its MustCall/CalledMethods
+
+import sys
+
+DEBUG = True
+
+
+def logout(x: str):
+    if (DEBUG):
+        print(x)
+
+
+class Field:
+    def __init__(self, field_name: str, field_index: int):
+        self.field_name = field_name
+        self.field_index = field_index
+
+
+class Struct:
+    def __init__(self, name: str):
+        self.name = name
+        self.typedefs: list[str] = []
+        self.fields: list[Field] = []
+
+    def get_name(self) -> str:
+        return self.name
+
+    def get_typedefs(self) -> "list[str]":
+        return self.typedefs
+
+    def add_field(self, field: Field) -> None:
+        self.fields.append(field)
+
+    def add_typedef(self, typedef: str) -> None:
+        self.typedefs.append(typedef)
+
+
+class StructsManager:
+    def __init__(self):
+        self.structs: list[Struct] = []
+
+    # if struct is not found, a new one is returned with name struct_name if error_if_not_found is false
+    def get_struct(self, struct_name: str, error_if_not_found: bool = False) -> Struct:
+        for struct in self.structs:
+            if struct.get_name() == struct_name or struct_name in struct.get_typedefs():
+                return struct
+
+        if error_if_not_found:
+            raise ValueError(f"struct {struct_name} not found")
+
+        new_struct = Struct(struct_name)
+        self.structs.append(new_struct)
+        return self.structs[-1]
+
+    def struct_exists(self, struct_name: str) -> bool:
+        for struct in self.structs:
+            if struct.get_name() == struct_name or struct_name in struct.get_typedefs():
+                return True
+
+        return False
+
+    # if there already exists a struct of name struct_name, that struct is returned.
+    # otherwise, the struct is added and returned.
+    def add_struct(self, struct_name: str) -> bool:
+        for struct in self.structs:
+            if struct.get_name() == struct_name:
+                return struct
+
+        new_struct = Struct(struct_name)
+        self.structs.append(new_struct)
+        return self.structs[-1]
+
+
+class StructVar:
+    def __init__(self, var_name: str, type_name: str):
+        self.var_name = var_name
+        self.type_name = type_name
+
+    def get_var_name(self):
+        return self.var_name
+
+    def get_type_name(self):
+        return self.type_name
+
+
+class StructVarManager:
+    def __init__(self):
+        self.struct_vars: list[StructVar] = []
+
+    # if there already exists a var of name var_name, that struct is returned.
+    # otherwise, the var is added and returned.
+    def add_struct_var(self, var_name: str, type_name: str) -> bool:
+        for struct_var in self.struct_vars:
+            if struct_var.get_var_name() == var_name:
```

Comment:
Rather than loop + if, could you use a `contains` method (or just `in`)? I think there's a more python-idiomatic ("pythonic"?) way to do this.

Reply:
Since this method checks for equality by comparing two different types, `StructVar` and `str` (`var_name` ), I feel defining a contains method for this specific case would be awkward